packages/amplify-cli/src/extensions/amplify-helpers/git-manager.ts

```diff
@@ -2,8 +2,12 @@ import * as fs from 'fs-extra';
 import * as os from 'os';
 import { LocalLogDirectory } from 'amplify-cli-logger';
 
-const amplifyMark = '#amplify';
+const amplifyMark = '#amplify-do-not-edit-begin';
+const amplifyEndMark = '#amplify-do-not-edit-end';
+const legacyAmplifyMark = '#amplify';
 const amplifyMarkRegExp = new RegExp(`^${amplifyMark}`);
+const amplifyEndMarkRegExp = new RegExp(`^${amplifyEndMark}`);
+const legacyAmplifyMarkRegExp = new RegExp(`^${legacyAmplifyMark}`);
```

Comment:
nit: I prefer "deprecated" over "legacy" (even though I've used it myself in the past)